crates/sui-framework/packages/sui-framework/sources/random.move

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#[allow(unused_use)]
+// This module provides functionality for generating and using secure randomness.
+//
+// Randomness is currently write-only, until user-facing API is implemented.
+module sui::random {
+    use std::vector;
+    use sui::dynamic_field;
+    use sui::object::{Self, UID};
+    use sui::transfer;
+    use sui::tx_context::{Self, TxContext};
+
+    // Sender is not @0x0 the system address.
+    const ENotSystemAddress: u64 = 0;
+    const EWrongInnerVersion: u64 = 1;
+
+    const CurrentVersion: u64 = 1;
+
+    /// Singleton shared object which stores the global randomness state.
+    /// The actual state is stored in a dynamic field of type RandomnessStateInner to support
+    /// future versions of the randomness state.
+    struct Random has key {
```

Comment:
Consider using versioned::Versioned directly, otherwise you may end up re-implementing a lot of same stuff

Reply:
@mystenmark curious before I migrate if there is any specific reason you didn't use this for authenticator state?

Reply:
I did not know about it :D

Reply:
Migrated